pxteditor/code-validation/runValidatorPlan.ts

```diff
@@ -72,8 +72,7 @@ function runBlocksExistValidation(usedBlocks: Blockly.Block[], inputs: pxt.block
     const blockResults = validateBlocksExist({ usedBlocks, requiredBlockCounts });
     let successfulBlocks: Blockly.Block[] = [];
     if (blockResults.passed) {
-        const blockIdsFromValidator = Object.keys(inputs.blockCounts)
-        const blockId = blockIdsFromValidator?.[0];
+        const blockId = inputs.blockCounts[0].blockId;
```

Comment:
blockCounts can have more than one entry, so I don't think we can assume index 0 here. Rather, we need to iterate over these ids. I also don't think we can assume index 0 in the line below.

Reply:
Yeah, this is something I thought of when I was first making this. I just kept it at the zeroth index because I hadn't seen a validator that we were building for the teacher tool that used multiple blocks inside the blocks exist. I thought that it would either go in a separate check or we would just use the block set validator instead. I can change this, though.

Your comment about the successful blocks makes me think, though. I'm not really sure why I made successful blocks a list.. Maybe just to keep with the pattern of the other things that we return from blocksExist since they all return lists. I'm thinking that it might be better to make successful blocks an Object instead, that way we don't _need_ to do any indexing, we can just do a look-up of `blockResults.successfulBlocks[blockId]` and this will return the list of successful blocks of that block id.